### PR TITLE
Migrate services tests to vitest

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

Migrate services tests to vitest.

# Changes

- in addition to the migration few `toastsShow` spy and test had to be improved. Either by adding a `waitFor` or few by testing the effective toast module function that is called. For example replacing `toastsShow` by `toastsSuccess` in a test.

### CODE REVIEW

File: frontend/src/vitests/lib/services/sns-neurons.services.spec.ts [hunk 1]
```diff
@@ -315,193 +312,198 @@ describe("sns-neurons-services", () => {
...
+        getSnsNeuron({
+          neuronIdHex: bytesToHexString(
+            Array.from(mockSnsNeuron.id[0]?.id as Uint8Array)
+          ),
+          rootCanisterId: mockPrincipal,
+          onLoad,
+        });
+      }));
+
+    it("should return neuron if it's in the store", (done) =>
```
Comment (lmuntaner) [inline frontend/src/vitests/lib/services/sns-neurons.services.spec.ts:415]: Not used

File: frontend/src/vitests/lib/services/sns-neurons.services.spec.ts [hunk 2]
```diff
@@ -315,193 +312,198 @@ describe("sns-neurons-services", () => {
...
+        getSnsNeuron({
+          neuronIdHex: bytesToHexString(
+            Array.from(mockSnsNeuron.id[0]?.id as Uint8Array)
+          ),
+          rootCanisterId: mockPrincipal,
+          onLoad,
+        });
+      }));
+
+    it("should call api even if it's in the store when forceFetch", (done) =>
```
Comment (lmuntaner) [inline frontend/src/vitests/lib/services/sns-neurons.services.spec.ts:447]: ditto

File: frontend/src/vitests/lib/services/sns-neurons.services.spec.ts [hunk 3]
```diff
@@ -315,193 +312,198 @@ describe("sns-neurons-services", () => {
...
+          forceFetch: true,
+          neuronIdHex: bytesToHexString(
+            Array.from(mockSnsNeuron.id[0]?.id as Uint8Array)
+          ),
+          rootCanisterId: mockPrincipal,
+          onLoad,
+        });
+      }));
+
+    it("should call onError callback when call failes", (done) =>
```
Comment (lmuntaner) [inline frontend/src/vitests/lib/services/sns-neurons.services.spec.ts:480]: ditto

File: frontend/src/vitests/lib/services/vote-registration.services.spec.ts [hunk 4]
```diff
@@ -109,37 +105,43 @@ describe("vote-registration-services", () => {
...
-          proposalInfo: proposal,
-          vote: Vote.Yes,
-          reloadProposalCallback: () => {
-            updateContextCalls += 1;
-            if (updateContextCalls === neuronIds.length) {
-              done();
-            }
-          },
-        });
+      it("should update store with a new vote registration", (done) =>
```
Comment (lmuntaner) [inline frontend/src/vitests/lib/services/vote-registration.services.spec.ts:108]: ditto, not used

**Review by lmuntaner — APPROVED** [review]
LGTM!

Revert the problematic spec files and we can work on them separately and the CR can be done with more detail.

File: frontend/src/vitests/lib/services/_public/sns-proposals.services.spec.ts [hunk 5]
```diff
@@ -257,113 +253,117 @@ describe("sns-proposals services", () => {
...
-              rootCanisterId,
-              proposalId,
-            });
-            expect(queryProposalSpy).toBeCalledTimes(1);
-            done();
-          },
-          reloadForBallots: false,
-        });
-      });
+      it("should use the proposal in store if certified and not call api", (done) =>
```
[the remark this answers is not in the document]
Comment (peterpeterparker) [reply frontend/src/vitests/lib/services/_public/sns-proposals.services.spec.ts:261]: Yeah it was just already the case I think, I'll remove it here then.

File: frontend/src/vitests/lib/services/sns-neurons.services.spec.ts (hunk 1 above)
Comment (peterpeterparker) [reply frontend/src/vitests/lib/services/sns-neurons.services.spec.ts]: Test reverted. Done already there on main.

File: frontend/src/vitests/lib/services/sns-neurons.services.spec.ts (hunk 2 above)
Comment (peterpeterparker) [reply frontend/src/vitests/lib/services/sns-neurons.services.spec.ts]: Test reverted. Done already there on main.

File: frontend/src/vitests/lib/services/sns-neurons.services.spec.ts (hunk 3 above)
Comment (peterpeterparker) [reply frontend/src/vitests/lib/services/sns-neurons.services.spec.ts]: Test reverted. Done already there on main.

File: frontend/src/vitests/lib/services/vote-registration.services.spec.ts (hunk 4 above)
Comment (peterpeterparker) [reply frontend/src/vitests/lib/services/vote-registration.services.spec.ts]: Removed now here

**Pull request merged**